### PR TITLE
HEE-230: Contact card

### PR DESCRIPTION
**Pull request by thojava**
- Add Person, Department document types
- Display the contact fields based on display options

### CODE REVIEW

File: repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/contactCard.ftl [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<#assign hst=JspTaglibs["http://www.hippoecm.org/jsp/hst/core"] >
```
Comment (smanivasag) [inline repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/contactCard.ftl:1]: @thojava - Could we keep dash based naming convention for contactCard like `contact-card.ftl` in order to be inline with other macros ?

**Review by smanivasag — COMMENTED** [review]
Hi @thojava - Thanks for the PR and it looks great 👍 

Few points:
- Could we get rid of `hee:image` field from `hee:person` DocumentType as it doesn't seem to be used in the implemenation ?
- Just wondering what happens in a scenario wherein for example `hee:phoneNumber` isn't actually available on the `hee:person` document, but it is checked on `hee:contactCard` document ? Shall we make the fields on `hee:person` DocumentType as mandatory ? Or shall we update the `contactCard.ftl` template to not to render those ? @MattBrooksManifesto - Thoughts ?
- @MattBrooksManifesto  - Does `Address` field needs to be a text area ?

File: repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/contactCard.ftl (hunk 1 above)
Comment (thojava) [reply repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/contactCard.ftl]: yeap!

**Comment by MattBrooksManifesto** [comment]
There is a scenario where Phone Number would not be on the Person object but requested on Contact Card, so we will need to accommodate that

**Comment by smanivasag** [comment]
> There is a scenario where Phone Number would not be populated on the Person object but requested on Contact Card, so we will need to accommodate that.
> 
> Address should be a textarea I think (doesn't need to be richtext)

Thanks @MattBrooksManifesto 

@thojava - I think we probably need to handle this on the `contact-card.ftl` to not render a field if it's unavailable in the person document. Hope that makes sense. Thanks!

**Comment by thojava** [comment]
I think we probably need to handle this on the contact-card.ftl to not render a field if it's unavailable in the person document. Hope that makes sense. Thanks!
@smanivasag I have made it can you review again ?

**Comment by smanivasag** [comment]
> I think we probably need to handle this on the contact-card.ftl to not render a field if it's unavailable in the person document. Hope that makes sense. Thanks!
> @smanivasag I have made it can you review again ?

@thojava - Looks good, thanks 👍 

Re. `hee:image` on the `hee:person` document - are we planning to render this as part of another ticket (e.g. HEE-218) ?

@MattBrooksManifesto - Just realised that making `Address` field as TextArea would still render the address in a single line on the page and @thojava will need to format the address replacing line breaks with `<br/>` tag to render the address fields to render in separate lines. Hope thats OK ? Thanks!

**Comment by MattBrooksManifesto** [comment]
That makes sense to me @smanivasag

**Comment by smanivasag** [comment]
Leaving `hee:image` field on `hee:person` DocumentType as is as we need this for `HEE-103` ticket.

Made the following changes as part of the latest commit:
- Converted `hee:address` from `Text` to `TextArea` field.
- Updated `contat-card.ftl` to render line breaks with `<br>` tag for Address.

**Review by smanivasag — APPROVED** [review]

**Pull request merged**